app/src/component/searchbar/SearchBar.js

```diff
@@ -137,18 +136,30 @@ class IntegrationAutosuggest extends React.Component {
     });
   };
 
-  handleChange = (event, { newValue }) => {
-    this.setState({
-      value: newValue,
-    });
-    if (checkAirportInList(newValue)) {
-      this.props.setAirport(newValue);
+  handleChange = (event, { newValue, method }) => {
+    if (method == 'type') {
+      this.setState({
+        value: newValue,
+      });
+    }
+  };
+
+  handleKeyPress = (event) => {
+    if (event.key === 'Enter') {
+      if (checkAirportInList(this.state.value)) {
+        this.props.setAirport(this.state.value.toUpperCase());
+        this.state.value = '';
```

Comment:
Always use this.setState to change value

Reply:
Sure, I will. Ref: https://stackoverflow.com/questions/35867038/what-the-difference-of-this-state-and-this-setstate-in-reactjs